soc/arm/nordic_nrf/common/soc_nrf_common.h

```diff
@@ -120,11 +120,18 @@
  *
  *     foo: my-node {
  *             tx-gpios = <&gpio0 4 ...>;
- *             rx-gpios = <&gpio1 5 ...>;
+ *             rx-gpios = <&gpio0 5 ...>, <&gpio1 5 ...>;
  *     };
  *
- *     NRF_DT_GPIOS_TO_PSEL(DT_NODELABEL(foo), tx_gpios) // 0 + 4 = 4
- *     NRF_DT_GPIOS_TO_PSEL(DT_NODELABEL(foo), rx_gpios) // 32 + 5 = 37
+ *     NRF_DT_GPIOS_TO_PSEL_BY_IDX(DT_NODELABEL(foo), tx_gpios, 0) // 0 + 4 = 4
+ *     NRF_DT_GPIOS_TO_PSEL_BY_IDX(DT_NODELABEL(foo), rx_gpios, 1) // 32 + 5 = 37
+ */
+#define NRF_DT_GPIOS_TO_PSEL_BY_IDX(node_id, prop, idx)	\
+	(DT_GPIO_PIN_BY_IDX(node_id, prop, idx) + \
+	(DT_PROP_BY_PHANDLE_IDX(node_id, prop, idx, port) << 5))
```

Comment:
NRF_GPIO_PIN_MAP(
  DT_PROP_BY_PHANDLE_IDX(node_id, prop, idx, port),
  DT_GPIO_PIN_BY_IDX(node_id, prop, idx)
  )

the same with the below macro please.

Reply:
Done for both macros